src/sql/platform/connection/common/connectionManagement.ts

```diff
@@ -333,6 +333,22 @@ export interface IConnectionManagementService {
 	 * @returns Promise with a boolean value indicating whether the user has accepted the suggestion.
 	 */
 	handleUnsupportedProvider(providerId: string): Promise<boolean>;
+
+	/**
+	 * Launches the password change dialog.
+	 * @param profile The connection profile to change the password.
+	 * @param params Required connection parameters associated with the profile.
+	 */
+	openChangePasswordDialog(profile: IConnectionProfile, params: INewConnectionParams): void;
+
+	/**
+	 * converts the connection profile into an RPC friendly type (azdata.connection.ConnectionProfile).
+	 * @param profile The connection profile to be converted.
+	 * @param removeCredentials Remove credentials from the profile.
+	 * @param deepCopyOptions Deep copy the options of the profile to the converted profile.
+	 * @returns a profile without any services to be passed via the RPC protocol
+	 */
+	convertToConnectionProfile(profile: IConnectionProfile, removeCredentials: boolean, deepCopyOptions: boolean): azdata.connection.ConnectionProfile;
```

Comment:
we shouldn't directly reference types defined in the azdata typing as it doesn't exist at runtime in ADS core. also exposing this method is weird to me, you shouldn't need the service to do it for you, at the most, you need a util function.

Reply:
Sigh, yeah this whole typings thing is a mess. In the way it's being used here it's technically fine but there's a couple of problems with doing something like this : 

1. It's defined as a class but the thing being created isn't an actual class - just an object with the same shape (so you couldn't actually use instanceof despite it seeming like you could)
2. The static function createFrom doesn't actually exist (not a problem for the objects being created specifically, but still weird and confusing)

At the very least this isn't a type we should be passing around in core code - it should only be used on the boundaries here. You should use one of the internal IConnectionProfile/ConnectionProfile types defined in core and then convert it into the azdata version if needed before sending to the extension host.

Reply:
@alanrenmsft what about the password removal function? It is required for some of the conversion calls, and it's implemented in connection Management Service, if we're going to move this to utils, we will have to remove it and place it outside the utils call.

Reply:
Just pass in the service as a parameter.

Reply:
I decided to pass in the function instead, which would do the same thing.